opentech/public/news/views.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.syndication.views import Feed
+
+from opentech.public.news.models import NewsPage, NewsType
+
+
+class NewsFeed(Feed):
+    link = "https://www.opentech.fund/news/"
+    title = "OTF – News, updates, and announcements"
+    description = "News, updates, and announcements from The Open Technology Fund"
+
+    def items(self):
+        return NewsPage.objects.live().order_by('-first_published_at')[:20]
+
+    def item_title(self, item):
+        return item.title
+
+    def item_description(self, item):
+        return item.body
+
+    def item_pubdate(self, item):
+        return item.display_date
+
+
+class NewsTypesFeed(Feed):
+    link = "https://www.opentech.fund/news/"
+
+    def get_object(self, request, type):
+        return NewsType.objects.get(id=type)
+
+    def title(self, type):
+        return f"OTF – News of type {type}"
+
+    def description(self, type):
+        return f"News, updates, and announcements of type {type} from The Open Technology Fund"
+
+    def items(self, type):
+        return NewsPage.objects.live().filter(news_types__news_type=type).order_by('-first_published_at')[:20]
```

Comment:
We are using `display_date` property for `pubDate`. Therefore can use [Coalesce](https://docs.djangoproject.com/en/2.2/ref/models/database-functions/#coalesce) to have the same behaviour.

 ```python
NewsPage.objects.live().public().filter(news_types__news_type=type).annotate(
            date=Coalesce('publication_date', 'first_published_at')).order_by('-date')[:20]
```

Reply:
Learned a new thing, thanks!